libr/util/str.c

```diff
@@ -248,14 +248,14 @@ R_API R_MUSTUSE char *r_str_r2_prefix(const char *str) {
 	return r_str_newf ("%s%s%s", r_sys_prefix (NULL), R_SYS_DIR, str);
 }
 
-// Compute a 64 bit DJB hash of a string.
+// Compute a 64 bit DJB hash of a string (See the XOR version in http://www.cse.yorku.ca/~oz/hash.html)
 R_API ut64 r_str_hash64(const char *s) {
 	ut64 len, h = 5381;
 	if (!s) {
 		return 0;
 	}
 	for (len = strlen (s); len > 0; len--) {
-		h = (h ^ (h << 5)) ^ *s++;
+		h = (h + (h << 5)) ^ *s++;
```

Comment:
this comment in sdb explains the reason for the change. 

<img width="1152" alt="Screenshot 2024-11-01 at 11 52 20 PM" src="https://github.com/user-attachments/assets/3a20fc15-486d-46b5-a85a-ce3bfdc5cd55">

Reply:
Ok I corrected it and add a comment to explain.

Reply:
i dont thikn the algo is the same,.. but in my tests it was behaving faster and having the same amount of collisions. but im not a mathematician so dunno

Reply:
Yes it changed with the carry propagation. But we can keep it like that.